Source/Tools/FEXInterpreter/ELFCodeLoader.h

```diff
@@ -707,6 +709,70 @@ class ELFCodeLoader final : public FEX::CodeLoader {
     *AuxTabSize = sizeof(AuxType) * AuxVariables.size();
   }
 
+  // Get the current memory map from /proc/self/stat
+  static bool GetCurrentMap(struct prctl_mm_map& map) {
+    int fd = open("/proc/self/stat", O_RDONLY | O_CLOEXEC);
+    if (fd < 0) {
+      return false;
+    }
+
+    // /proc/self/stat has 52 fields of at most 20 digits each (UINT64_MAX).
+    // 52*20 = 1040, so 2048 is a conservative upper bound
+    char stat_buffer[2048];
+    ssize_t bytes_read = read(fd, stat_buffer, sizeof(stat_buffer) - 1);
+    close(fd);
+
```

Comment:
We have a helper to load a whole file into a buffer:
```suggestion
    // /proc/self/stat has 52 fields of at most 20 digits each (UINT64_MAX).
    // 52*20 = 1040, so 2048 is a conservative upper bound
    char stat_buffer[2048];
    auto bytes_read = LoadFileToBuffer("/proc/self/stat", stat_buffer);
```